### DISCUSSION TITLE
Queuing and retries

### DISCUSSION THREAD

**Discussion by cuu508** [decke/smtprelay#66] [Ideas]
Would message queuing and retries be in scope for this project?

Suppose the remote SMTP server is having issues (no route to host, connection rejected, connection timeout, SSL certificate problems, "internal error" type error messages). In these cases it would be nice to retry delivery a little bit later. I know [Postfix can do it](https://docs.emaillabs.io/get_started/smtp-relay-configuration/) but am looking at more lightweight alternatives like dma, ssmtp, msmtp, nullmailer and smtprelay. But AFAICT none of them currently supports retries.

**Comment by decke** [comment]
Queueing is definitely out of scope for smtprelay because that would mean it needs to be rewritten to store and forward.

The good news is that at least dma can do retries and probably others as well. dma stores the mail temporary in /var/spool/dma

**Reply by cuu508** [reply]
Thanks – understood! Thanks for the pointer about dma, I will look into it!
